python/tvm/relay/dataflow_pattern/__init__.py

```diff
@@ -748,11 +756,14 @@ def rewrite(callbacks, expr: Expr) -> Expr:
         The Expression with matched subgraphs rewritten by the callbacks.
     """
     if isinstance(callbacks, DFPatternCallback):
-        tmp = [_DFPatternCallback(callbacks.pattern, callbacks.callback)]
+        assert callbacks.pattern is not None
+        tmp = [_DFPatternCallback(callbacks.pattern, callbacks.callback, callbacks.require_type)]
     else:
         tmp = []
         for callback in callbacks:
-            tmp.append(_DFPatternCallback(callback.pattern, callback.callback))
+            assert callback.pattern is not None
+            tmp.append(_DFPatternCallback(callback.pattern, callback.callback,
+                                          callback.require_type))
```

Comment:
We can simplify this part by adding
`callbacks = [callbacks] if instance(callbacks, DFPatternCallback) else callback`
on the top.

Reply:
Fixed